--- ibpp-2.3.5.0/array.cpp.orig	2004-09-23 17:34:08.000000000 +0200
+++ ibpp-2.3.5.0/array.cpp	2004-10-31 11:56:34.577764120 +0100
@@ -289,7 +289,7 @@
 		throw ExceptionImpl("Array::ReadTo", "Wrong count of array elements");
 
 	IBS status;
-	long lenbuf = mBufferSize;
+	ISC_LONG lenbuf = mBufferSize;
 	(*gds.Call()->m_array_get_slice)(status.Self(), mDatabase->GetHandlePtr(),
 		mTransaction->GetHandlePtr(), &mId, &mDesc, mBuffer, &lenbuf);
 	if (status.Errors())
@@ -1088,7 +1088,7 @@
 	}
 
 	IBS status;
-	long lenbuf = mBufferSize;
+	ISC_LONG lenbuf = mBufferSize;
 	(*gds.Call()->m_array_put_slice)(status.Self(), mDatabase->GetHandlePtr(),
 		mTransaction->GetHandlePtr(), &mId, &mDesc, mBuffer, &lenbuf);
 	if (status.Errors())
